src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/TeddyHelper.cs

```diff
@@ -323,24 +323,14 @@ private static Vector128<byte> RightShift1(Vector128<byte> left, Vector128<byte>
             // We want to shift the last element of left (15) to be the first element of the result
             // result: [15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]
 
-            if (Ssse3.IsSupported)
-            {
-                return Ssse3.AlignRight(right, left, 15);
-            }
-            else
-            {
-                // We create a temporary 'leftShifted' vector where the 1st element is the 16th element of the input.
-                // We then use TBX to shuffle all the elements one place to the left.
-                // 0xFF is used for the first element to replace it with the one from 'leftShifted'.
-
-                Vector128<byte> leftShifted = Vector128.Shuffle(left, Vector128.Create(15, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0).AsByte());
-                return AdvSimd.Arm64.VectorTableLookupExtension(leftShifted, right, Vector128.Create(0xFF, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14));
-            }
+            return Ssse3.IsSupported
+                ? Ssse3.AlignRight(right, left, 15)
+                : AdvSimd.ExtractVector128(left, right, 15);
```

Comment:
It's worth noting that...

`AlignRight` is exposed as `(left, right, mask)` where `left` is the `upper` 128-bits and `right` is the lower 128-bits of the concatenated vector, therefore mask `1` picks `right[1]-right[15]` and `left[0]` as the last element.

`AdvSimd.ExtractVector128` is exposed as `(upper, lower, mask)`, but it should've been `(lower, upper, mask)`, so its inverse from the `AlignRight` order. This can be potentially confusing when reviewing the code.

The PR here is doing it correctly, just wanted to call out that its a weird thing if you go look at stuff